packages/node_modules/@webex/plugin-meetings/src/stats/events.js

```diff
@@ -0,0 +1,136 @@
+import {StatelessSparkPlugin} from '@ciscospark/spark-core';
+
+import {DEFAULT_EVENT_VIDEO_SEND_KEYS, DEFAULT_EVENT_AUDIO_SEND_KEYS, DEFAULT_EVENT_AUDIO_RECEIVE_KEYS, DEFAULT_EVENT_VIDEO_RECEIVE_KEYS, EVENT_TYPES, EVENT_TRIGGERS, EVENT_STATS_MAP, MEETINGS, AUDIO, VIDEO} from '../constants';
+
+/**
+ * @param {Object} operate the filtered, parsed, converted, transformed, simplified data point to check against
+ * @param {String} type - local or remote
+ * @param {String} keys - the stat keys for types of stats defined by DEFAULT_TRANSFORM_REGEX
+ * @param {String} stat - the accessor to get the actual stat
+ * @param {String} kind - audio or video
+ * @returns {Object} always whatever the first sentFirstVideoBytes were, in the past or if it happened now, or undefined if never
+ */
+const operateEvent = (operate, type, keys, stat, kind) => {
+  const something = {};
+  something.found = keys.some((key) => {
+    if (operate[key] && operate[key][stat] && operate[key][stat] > 0) {
+      something.match = {
+        type,
+        key,
+        data: operate[key],
+        stat,
+        kind
+      };
+      return true;
+    }
+    return false;
+  });
+  return something;
+};
+
+export default class StatsEvents extends StatelessSparkPlugin {
+  namespace = MEETINGS;
+
+  /**
+   * constructs an instance
+   * @param {Object} attrs
+   * @param {Object} options
+   */
+  constructor(attrs, options) {
+    super({}, options);
+    this.sentFirstVideoBytes = undefined;
+    this.receivedFirstVideoBytes = undefined;
+    this.sentFirstAudioBytes = undefined;
+    this.receivedFirstAudioBytes = undefined;
```

Comment:
I typically prefer to explicitly set variables to `null` instead of defining them as `undefined`

Reply:
hmmm we can change this whole sale in the app later